src/com/company/IntListImpl.java

```diff
@@ -0,0 +1,232 @@
+package com.company;
+
+import java.util.Arrays;
+
+public class IntListImpl<E> implements IntList{
+
+    private Integer[] array = new Integer[10];
+
+    private int size = 0;
+    public IntListImpl(Integer[] array, int size) {
+        this.array = array;
+        this.size = size;
+    }
+
+    public Integer[] getArray() {
+        return array;
+    }
+
+    public int getSize() {
+        return size;
+    }
+
+    public IntListImpl<Integer> generateArray() {
+        return generateRandomArray();
+    }
+
+    private static IntListImpl<Integer> generateRandomArray() {
+        java.util.Random random = new java.util.Random();
+        Integer[] arr = new Integer[100_000];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = random.nextInt(100_000) + 100_000;
+        }
+        return new IntListImpl<Integer>(arr,arr.length);
+    }
+
+    @Override
+    public Integer add(Integer item) {
+        if (size == array.length) {
+            return Integer.parseInt("Массив уже заполнен элементами");
+        }
+        array[size] = item;
+        size++;
+        return array[size];
+    }
+
+    @Override
+    public Integer add(int index, Integer item) {
+        if (index >= size) {
+            throw new IndexLargerSizeOfArrayException("Значение индекса " + index + " больше чем размер массива!");
+        } else {
+            array[index] = item;
```

Comment:
```suggestion
            System.arraycopy(array, index, array, index + 1, size - index);
            array[index] = item;
```
следует выполнить сдвиг, чтобы не потерять элементы по текущему индексу. Add должен добавить, при этом не потерять существующий элемент в списке